src/main/kotlin/com/quipalup/katydid/logentry/primaryadapter/httprest/CreateLogEntryEndpoint.kt

```diff
@@ -1,29 +1,36 @@
 package com.quipalup.katydid.logentry.primaryadapter.httprest
 
+import arrow.core.Either
+import arrow.core.flatMap
+import arrow.core.right
+import com.quipalup.katydid.logentry.application.CreateLogEntryByFieldCommand
+import com.quipalup.katydid.logentry.application.CreateLogEntryCommandHandler
+import com.quipalup.katydid.logentry.domain.LogEntryError
 import org.springframework.http.HttpStatus
 import org.springframework.web.bind.annotation.PostMapping
-import org.springframework.web.bind.annotation.RequestBody
 import org.springframework.web.bind.annotation.ResponseStatus
 import org.springframework.web.bind.annotation.RestController
 
 @RestController
-class CreateLogEntryEndpoint {
+internal class CreateLogEntryEndpoint(private val createLogEntryCommandHandler: CreateLogEntryCommandHandler) {
     @PostMapping("/log-entries")
     @ResponseStatus(HttpStatus.CREATED)
-    fun execute(@RequestBody newLog: LogEntryRequestDocument): LogEntryResponseDocument {
+    fun execute(): LogEntryResponseDocument {
```

Comment:
```suggestion
    fun execute(@RequestBody newLog: LogEntryRequestDocument): LogEntryResponseDocument {
```
That is was Ok. This endpoint will receive a JSON and it will be deserialised by Spring into this parameter. That data is which we will have to transform into a command and send it to the command handler.